### PR TITLE
Adding tests

### PR DESCRIPTION
**Pull request by PavelApis**

### CODE REVIEW

File: src/test/java/io/github/pavelapis/minesweeper/FieldTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                {new Cell(2, 0, -1), new Cell(2, 1, 0), new Cell(2, 2, 0)}
+        };
+        Field field = new Field(cells, 3);
+        field.setCellsValues();
+        int[][] expectedResult = {
+                {-1, 2, -1},
+                {2, 3, 1},
+                {-1, 1, 0}
+        };
+        for (int i = 0; i < 3; i++){
```
Comment (mfvanek) [inline src/test/java/io/github/pavelapis/minesweeper/FieldTest.java:52]: Looks like checkstyle isn't working

File: src/test/java/io/github/pavelapis/minesweeper/FieldTest.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            -1 0 -1               -1 2 -1
+             0 0 0  -must become-> 2 3 1
+            -1 0 0                -1 1 0
+         */
+    @Test
+    void settingCellsValue(){
+        Cell[][] cells = {
+                { new Cell(0, 0, -1), new Cell(0, 1, 0), new Cell(0, 2, -1)},
+                { new Cell(1, 0, 0), new Cell(1, 1, 0), new Cell(1, 2, 0)},
+                {new Cell(2, 0, -1), new Cell(2, 1, 0), new Cell(2, 2, 0)}
```
Comment (mfvanek) [inline src/test/java/io/github/pavelapis/minesweeper/FieldTest.java:43]: Looks like checkstyle isn't working

File: src/main/java/io/github/pavelapis/minesweeper/gameclasses/IconMaker.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+public class IconMaker {
+
+    public static void setIcon(final JButton button, final String name) {
+        try (InputStream stream = Thread.currentThread().getContextClassLoader()
+                .getResourceAsStream(name + ".jpg")) {
+            final BufferedImage bufferedImage = ImageIO.read(stream);
+            button.setIcon(new ImageIcon(bufferedImage));
+        } catch (IOException e) {
+            final Logger log = Logger.getLogger(IconMaker.class.getName());
+            if (log.isLoggable(Level.INFO)) {
```
[the remark this answers is not in the document]
Comment (PavelApis) [reply src/main/java/io/github/pavelapis/minesweeper/gameclasses/IconMaker.java:21]: Change to ` if( log,isInfoEnabled())`

File: src/test/java/io/github/pavelapis/minesweeper/CellTest.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import io.github.pavelapis.minesweeper.gameclasses.Cell;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+
+public class CellTest {
+    @Test
+    public void cellsCreation(){
+        Cell cell = new Cell(1, 2);
+        assertThat(cell.getValue()).isEqualTo(0);
```
[the remark this answers is not in the document]
Comment (PavelApis) [reply src/test/java/io/github/pavelapis/minesweeper/CellTest.java:12]: Changed it

File: src/test/java/io/github/pavelapis/minesweeper/CellTest.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
+package io.github.pavelapis.minesweeper;
+
+import io.github.pavelapis.minesweeper.gameclasses.Cell;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+
+public class CellTest {
```
[the remark this answers is not in the document]
Comment (PavelApis) [reply src/test/java/io/github/pavelapis/minesweeper/CellTest.java:8]: Done

File: src/test/java/io/github/pavelapis/minesweeper/FieldTest.java (hunk 1 above)
Comment (PavelApis) [reply src/test/java/io/github/pavelapis/minesweeper/FieldTest.java:52]: Now checkstyle parse the tests

File: config/checkstyle/checkstyle.xml [hunk 6]
```diff
@@ -172,7 +172,7 @@
                       value="^org\.springframework\.lang\.(NonNull|Nullable), ^org\.jetbrains\.annotations\.(NotNull|Nullable),
                 ^io\.micrometer\.core\.lang\.(NonNull|Nullable),
                 ^org\.elasticsearch\.common\.Nullable, ^org\.apache\.thrift\.annotation\.Nullable,
-                ^org\.junit\.Test, ^org\.junit\.jupiter\.api\.Assertions, ^org\.junit\.Test, ^org\.junit\.jupiter\.api\.Assertions\..*"/>
```
Comment (mfvanek) [inline config/checkstyle/checkstyle.xml:175]: Just revert these changes

File: src/main/java/io/github/pavelapis/minesweeper/gameclasses/IconMaker.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+import java.io.InputStream;
+import javax.imageio.ImageIO;
+import javax.swing.*;
+
+@Slf4j
```
Comment (mfvanek) [inline src/main/java/io/github/pavelapis/minesweeper/gameclasses/IconMaker.java:11]: Duplicated annotations

File: src/main/java/io/github/pavelapis/minesweeper/gameclasses/IconMaker.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+@Slf4j
+public class IconMaker {
+
+    public static void setIcon(final JButton button, final String name) {
+        try (InputStream stream = Thread.currentThread().getContextClassLoader()
+                .getResourceAsStream(name + ".jpg")) {
+            final BufferedImage bufferedImage = ImageIO.read(stream);
+            button.setIcon(new ImageIcon(bufferedImage));
+        } catch (IOException e) {
```
Comment (mfvanek) [inline src/main/java/io/github/pavelapis/minesweeper/gameclasses/IconMaker.java:20]:
```suggestion
        } catch (IOException e) {
              log.error(e);
        }
```

File: src/test/java/io/github/pavelapis/minesweeper/CellTest.java [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+import io.github.pavelapis.minesweeper.gameclasses.Cell;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+
+class CellTest {
+
+    protected CellTest() {
```
Comment (mfvanek) [inline src/test/java/io/github/pavelapis/minesweeper/CellTest.java:11]: Remove constructor

File: src/test/java/io/github/pavelapis/minesweeper/gameclasses/CellTest.java [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    /* default */ void testIllegalConstructorArguments() {
+        assertThatThrownBy(() -> new Cell(-1, 2)).isInstanceOf(IllegalArgumentException.class);
+        assertThatThrownBy(() -> new Cell(2, -1)).isInstanceOf(IllegalArgumentException.class);
+        assertThatThrownBy(() -> new Cell(2, 3, -10)).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+        /* default */ void cellsCreation() {
+        final Cell cell = new Cell(1, 2);
+        assertThat(cell.getValue()).isEqualTo(0);
```
Comment (mfvanek) [inline src/test/java/io/github/pavelapis/minesweeper/gameclasses/CellTest.java:23]: Replace all .isEqualTo(0) with .isZero()

File: .github/workflows/build.yml [hunk 11]
```diff
@@ -21,5 +21,14 @@ jobs:
           java-version: '17'
           distribution: 'temurin'
           cache: maven
-      - name: Build with Maven
+      - name: Build with Maven Ubuntu
+        if: matrix.os == 'ubuntu-latest'
+        run: xvfb-run mvn clean install
+      - name: Build with Maven Windows
```
Comment (mfvanek) [inline .github/workflows/build.yml:27]: it's better not to duplicate code. combine windows and macos runs together

File: src/test/java/io/github/pavelapis/minesweeper/gameclasses/CellTest.java [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+
+class CellTest {
+
+    /*
+        Cell constructor with illegal arguments must throw IllegalArgumentException
+     */
+    @Test
+    /* default */ void testIllegalConstructorArguments() {
```
Comment (mfvanek) [inline src/test/java/io/github/pavelapis/minesweeper/gameclasses/CellTest.java:15]: please remove all `/* default */`

File: src/test/java/io/github/pavelapis/minesweeper/gameclasses/CellTest.java [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+        assertThatThrownBy(() -> new Cell(-1, 2)).isInstanceOf(IllegalArgumentException.class);
+        assertThatThrownBy(() -> new Cell(2, -1)).isInstanceOf(IllegalArgumentException.class);
```
Comment (mfvanek) [inline src/test/java/io/github/pavelapis/minesweeper/gameclasses/CellTest.java:17]: also add assert on exception message via `hasMessage`

File: README.md [hunk 14]
```diff
@@ -1,8 +1,9 @@
 # MineSweeper -  Java Swing
 [![Java CI](https://github.com/PavelApis/MineSweeper/actions/workflows/build.yml/badge.svg)](https://github.com/PavelApis/MineSweeper/actions/workflows/build.yml "Java CI")
 ![version](https://img.shields.io/badge/version-1.0.0-yellow)
+[![codecov](https://codecov.io/gh/PavelApis/MineSweeper/branch/master/graph/badge.svg?token=R5LJA2BHTO)](https://codecov.io/gh/PavelApis/MineSweeper)
```
Comment (mfvanek) [inline README.md:4]: Remove token` token=R5LJA2BHTO`

**Review by mfvanek — DISMISSED** [review]
